core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/mercury/mercury.go

```diff
@@ -117,6 +110,69 @@ func (l *StreamsLookup) IsMercuryV03() bool {
 	return l.FeedParamKey == FeedIDs
 }
 
-func (l *StreamsLookup) IsMercuryUsingBatchPathV03() bool {
+func (l *StreamsLookup) IsMercuryV03UsingBlockNumber() bool {
 	return l.TimeParamKey == BlockNumber
 }
```

Comment:
this feels a little confusing to me, because both this function and `IsMercuryV02()` could return true at the same time. I would rename to just `IsUsingBlockNumber()` or something that doesn't imply it's using mercury v0.3.

Reply:
> both this function and IsMercuryV02() could return true at the same time.
This is true. 

The reason we want to have MercuryV03 in the name is because this function is used only by MercuryV03 to determine the batch path. We want to be clear that V02 cant use this function.

Reply:
Discussed offline, added a comment to explain this code.